auditbeat/module/auditd/audit_linux.go

```diff
@@ -137,10 +137,32 @@ func newAuditClient(c *Config, log *logp.Logger) (*libaudit.AuditClient, error)
 	return libaudit.NewAuditClient(nil)
 }
 
+func closeAuditClient(client *libaudit.AuditClient) error {
+	discard := func(bytes []byte) ([]syscall.NetlinkMessage, error) {
+		return nil, nil
+	}
+	// Drain the netlink channel in parallel to Close() to prevent a deadlock.
+	// This goroutine will terminate once receive from netlink errors (EBADF,
+	// EBADFD, or any other error). This happens because the fd is closed.
+	go func() {
+		for {
+			_, err := client.Netlink.Receive(true, discard)
```

Comment:
This is the only reader at this point, correct? Do we have to worry about any sort of data races in the case of a successfully started client that also has a read loop, or we don't care because we're shutting down anyway?

Reply:
There is the main reader from auditd that may be reading at the same time. In my tests, there's no issue of having both readers in parallel